pandas/core/arrays/datetimelike.py

```diff
@@ -27,13 +28,55 @@
 
 from pandas.core.algorithms import checked_add_with_arr, take, unique1d
 import pandas.core.common as com
+from pandas.core.nanops import nanstd
 
 from pandas.tseries import frequencies
 from pandas.tseries.offsets import DateOffset, Tick
 
 from .base import ExtensionOpsMixin
 
 
+def _get_reduction_vals(obj, skipna):
+    if not len(obj):
+        return NaT
+
+    if obj.hasnans:
+        if not skipna:
+            return NaT
+        vals = obj.asi8[~obj._isnan]
+    else:
+        vals = obj.asi8
+    return vals
+
+
+def _make_reduction(op, diff=False, only_timedelta=False):
+    """
+    Make a unary reduction method that handles NaT appropriately.
+    """
+
+    def method(self, skipna=True, **kwargs):
+        if only_timedelta:
+            raise TypeError('"{meth}" reduction is not valid for {cls}'
```

Comment:
I don't understand this. `only_timedelta` makes me think we shouldn't raise here if the array is timedelta dtype, but right now it looks like we raise unconditionally.

Reply:
woops, that should be `if only_timedelta and not is_timedelta64_dtype(self)`